dvc/repo/experiments/queue/base.py

```diff
@@ -569,11 +574,12 @@ def match_queue_entry_by_name(
             if exp_name in entry_name_dict:
                 result[exp_name] = entry_name_dict[exp_name]
                 continue
-            for rev, entry in entry_rev_list:
-                if rev.startswith(exp_name):
-                    result[exp_name] = entry
-                    break
-            else:
+            if len(exp_name) >= 4:
+                for rev, entry in entry_rev_list:
+                    if rev.startswith(exp_name):
+                        result[exp_name] = entry
+                        break
+            if exp_name not in result:
```

Comment:
This string length comparison isn't really safe for us to do. We should really be using `resolve_rev` here instead of using the min length + `startswith` check.

Reply:
One other alternative here would be use scmrepo's `Git.is_sha` (and keep the `startswith` check):

```suggestion
            if self.scm.is_sha(exp_name):
                for rev, entry in entry_rev_list:
                    if rev.startswith(exp_name):
                        result[exp_name] = entry
                        break
            if exp_name not in result:
```